watershed_workflow/triangulation.py

```diff
@@ -299,6 +306,20 @@ def refine(vertices, area):
     return refine
 
 
+def refine_stream_triangles(river_corrs):
+    """Returns a refinement function for triangles that have all three vertices on stream mesh."""
+    
+    def refine(vertices, area):
```

Comment:
Here is another place where you really want to form the shape outside of the function and then call it repeatedly, not form the shape inside the function.

```
riv_corr = shapely.ops.unary_union(river_corrs).buffer(1)
def refine(verts, area):
    return all(riv_corr.intersects(shapely.geometry.Point(p)) for p in vertices)
```

should be sufficient.